From 216117519ba395973a60deca40eb54cf5f6b6516 Mon Sep 17 00:00:00 2001
From: Jackie Huang <jackie.huang@windriver.com>
Date: Wed, 21 Jun 2023 03:12:42 -0400
Subject: [PATCH] ingress-nginx: add multi arch support

Use DEB_HOST_ARCH instead of hard-coded 'amd64' in the rules,
which adds multi arch support, arm64 will be supported for now.

Test Plan:
PASS: build-pkgs on x86-64 host
PASS: build-image on x86-64 host
PASS: build-pkgs on arm64 host
PASS: build-image on arm64 host
PASS: apply the app on x86-64 target
PASS: apply the app on arm64 target

Story: 2010739
Task: 48283

Signed-off-by: Jackie Huang <jackie.huang@windriver.com>
Change-Id: Ib325520a52b5264ca568116667fab91652a8da65
---
 .../upstream/ingress-nginx-helm/debian/deb_folder/rules     | 6 ++++++
 stx-nginx-ingress-controller-helm/debian/deb_folder/rules   | 5 +++++
 .../ingress-nginx/ingress-nginx-static-overrides.yaml       | 2 +-
 3 files changed, 12 insertions(+), 1 deletion(-)

diff --git a/helm-charts/upstream/ingress-nginx-helm/debian/deb_folder/rules b/helm-charts/upstream/ingress-nginx-helm/debian/deb_folder/rules
index 1b3d91a..247d32b 100644
--- a/helm-charts/upstream/ingress-nginx-helm/debian/deb_folder/rules
+++ b/helm-charts/upstream/ingress-nginx-helm/debian/deb_folder/rules
@@ -2,6 +2,7 @@
 # export DH_VERBOSE = 1
 
 export DEB_VERSION = $(shell dpkg-parsechangelog | egrep '^Version:' | cut -f 2 -d ' ')
+export DEB_HOST_ARCH = $(shell dpkg-architecture -qDEB_HOST_ARCH 2>/dev/null)
 export PATCH_VERSION = $(shell echo $(DEB_VERSION) | cut -f 4 -d '.')
 export CHART_BASE_VERSION = $(shell echo $(DEB_VERSION) | sed 's/-/./' | cut -d '.' -f 1-3)
 export CHART_VERSION = $(CHART_BASE_VERSION)+STX.$(PATCH_VERSION)
@@ -18,6 +19,11 @@ override_dh_auto_build:
 	# Copy ingress-nginx charts
 	cp -r charts/* ingress-nginx
 
+	# Fix the hardcoded image name: defaultbackend-amd64
+	sed -i 's/amd64/$(DEB_HOST_ARCH)/g' \
+		ingress-nginx/ingress-nginx/values.yaml \
+		ingress-nginx/ingress-nginx/README.md
+
 	cp Makefile ingress-nginx
 
 	cd ingress-nginx && make CHART_VERSION=$(CHART_VERSION) ingress-nginx
diff --git a/stx-nginx-ingress-controller-helm/debian/deb_folder/rules b/stx-nginx-ingress-controller-helm/debian/deb_folder/rules
index 20abb50..d92278d 100755
--- a/stx-nginx-ingress-controller-helm/debian/deb_folder/rules
+++ b/stx-nginx-ingress-controller-helm/debian/deb_folder/rules
@@ -5,6 +5,7 @@ export ROOT = debian/tmp
 export APP_FOLDER = $(ROOT)/usr/local/share/applications/helm
 
 export DEB_VERSION = $(shell dpkg-parsechangelog | egrep '^Version:' | cut -f 2 -d ' ')
+export DEB_HOST_ARCH = $(shell dpkg-architecture -qDEB_HOST_ARCH 2>/dev/null)
 export RELEASE = $(shell cat /etc/build.info | grep SW_VERSION | cut -d'"' -f2)
 export REVISION = $(shell echo $(DEB_VERSION) | cut -f 4 -d '.')
 
@@ -23,6 +24,10 @@ override_dh_auto_build:
 	mkdir -p $(STAGING)
 	cp files/metadata.yaml $(STAGING)
 	cp -Rv fluxcd-manifests $(STAGING)
+
+	sed -i 's/@ARCH@/$(DEB_HOST_ARCH)/g' \
+		$(STAGING)/fluxcd-manifests/ingress-nginx/ingress-nginx-static-overrides.yaml
+
 	mkdir -p $(STAGING)/charts
 	cp /usr/lib/helm/*.tgz $(STAGING)/charts
 
diff --git a/stx-nginx-ingress-controller-helm/stx-nginx-ingress-controller-helm/fluxcd-manifests/ingress-nginx/ingress-nginx-static-overrides.yaml b/stx-nginx-ingress-controller-helm/stx-nginx-ingress-controller-helm/fluxcd-manifests/ingress-nginx/ingress-nginx-static-overrides.yaml
index e5f3384..8aa9538 100644
--- a/stx-nginx-ingress-controller-helm/stx-nginx-ingress-controller-helm/fluxcd-manifests/ingress-nginx/ingress-nginx-static-overrides.yaml
+++ b/stx-nginx-ingress-controller-helm/stx-nginx-ingress-controller-helm/fluxcd-manifests/ingress-nginx/ingress-nginx-static-overrides.yaml
@@ -78,7 +78,7 @@ controller:
 defaultBackend:
   image:
     digest: ""
-    repository: registry.k8s.io/defaultbackend-amd64
+    repository: registry.k8s.io/defaultbackend-@ARCH@
     tag: "1.5"
   nodeSelector:
     node-role.kubernetes.io/control-plane: ""
-- 
2.30.2

